src/plugins/kibana/server/routes/api/index_pattern_endpoints.js

```diff
@@ -0,0 +1,258 @@
+        .then(function parseResults(results) {
+          const hits = results.hits.hits;
+          return _.map(hits, (patternHit) => {
+            patternHit._source.fields = JSON.parse(patternHit._source.fields);
+            return patternHit._source;
+          });
+        })
+        .then((patterns) => {
+          return Promise.map(patterns, (pattern) => {
+            return getMappings(pattern.title, req).catch(() => {
+              return {};
+            }).then((mappings) => {
+              return stitchPatternAndMappings(pattern, mappings);
+            });
+          });
+        })
+        .then(removeDeprecatedFieldProps)
+        .then((patterns) => {
+          reply(patterns);
+        }, function (error) {
+          reply(handleESError(error));
+        });
+    }
+  });
+
+  server.route({
+    path: '/api/kibana/index_patterns/{id}',
+    method: 'GET',
+    handler: function (req, reply) {
+      const callWithRequest = server.plugins.elasticsearch.callWithRequest;
+      let pattern = req.params.id;
+
+      const params = {
+        index: '.kibana',
+        type: 'index-pattern',
+        id: req.params.id
+      };
+
+      Promise.join(
+        callWithRequest(req, 'get', params)
+          .then((result) => {
+            result._source.fields = JSON.parse(result._source.fields);
+            return result._source;
+          }),
+        getMappings(pattern, req),
+        stitchPatternAndMappings)
+        .then(removeDeprecatedFieldProps)
+        .then(function (pattern) {
+          reply(_.isArray(pattern) ? pattern[0] : pattern);
+        }, function (error) {
+          reply(handleESError(error));
+        });
+    }
+  });
+
+  server.route({
+    path: '/api/kibana/index_patterns',
+    method: 'POST',
+    handler: function (req, reply) {
+      if (_.isEmpty(req.payload)) { return reply(Boom.badRequest('Payload required')); }
+      const validation = Joi.validate(req.payload, indexPatternSchema.post);
+      if (validation.error) {
+        return reply(Boom.badRequest(validation.error));
+      }
+
+      const callWithRequest = server.plugins.elasticsearch.callWithRequest;
+      const indexPattern = _.cloneDeep(req.payload);
+      const isWildcard = _.contains(indexPattern.title, '*');
+      const mappings = _(req.payload.fields)
+        .indexBy('name')
+        .mapValues(value => value.mapping)
+        .omit(_.isUndefined)
+        .value();
+      indexPattern.fields = JSON.stringify(_.map(indexPattern.fields, (field) => {
+        return _.omit(field, 'mapping');
+      }));
+
+      const patternCreateParams = {
+        index: '.kibana',
+        type: 'index-pattern',
+        id: indexPattern.title,
+        body: indexPattern
+      };
+
+      callWithRequest(req, 'create', patternCreateParams)
+        .then((patternResponse) => {
+          if (!isWildcard || _.isEmpty(mappings)) {
+            return patternResponse;
+          }
+          else {
```

Comment:
Please convert this to use an early return rather than wrapping everything in an else branch.